VelocitySDK/Core/VLTRecordingConfig.m

```diff
@@ -18,17 +19,30 @@ @interface VLTRecordingConfig ()
 
 @implementation VLTRecordingConfig
 
-- (instancetype)initWithDictionary:(NSDictionary *)dictionary
++ (instancetype)configWithDictionary:(NSDictionary *)dictionary error:(NSError **)error
 {
-    self = [super init];
-    if (self) {
-        _sampleSize = [dictionary[@"sample_size"] doubleValue];
-        _captureInterval = [dictionary[@"capture_interval"] doubleValue];
-        _detectMotionOn = [dictionary[@"detect_motion_enabled"] boolValue];
+    NSTimeInterval sampleSize = [dictionary[@"sample_size"] doubleValue];
+    NSTimeInterval captureInterval = [dictionary[@"capture_interval"] doubleValue];
+    BOOL detectMotionOn = [dictionary[@"detect_motion_enabled"] boolValue];
+
+    if (sampleSize > 0 && captureInterval > 0) {
+        VLTRecordingConfig *config = [[VLTRecordingConfig alloc] initSampleSize:sampleSize
+                                                                       interval:captureInterval
+                                                               detectioMotionOn:detectMotionOn];
+        return config;
     }
-    return self;
+
+    if (error != NULL) {
```

Comment:
Yours will work, although from types perspective, it should be a comparison with `NULL`.